engines/freescape/language/instruction.cpp

```diff
@@ -0,0 +1,394 @@
+/* ScummVM - Graphic Adventure Engine
+ *
+ * ScummVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the COPYRIGHT
+ * file distributed with this source distribution.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+// Based on Phantasma code by Thomas Harte (2013)
+
+#include "freescape/freescape.h"
+#include "freescape/language/8bitDetokeniser.h"
+
+namespace Freescape {
+
+FCLInstruction::FCLInstruction(Token::Type _type) {
+	source = 0;
+	destination = 0;
+	additional = 0;
+	// TODO: learn modern constructor syntax
+	type = _type;
+	thenInstructions = nullptr;
+	elseInstructions = nullptr;
+}
+
+FCLInstruction::FCLInstruction() {
+	source = 0;
+	destination = 0;
+	additional = 0;
+	type = Token::UNKNOWN;
+	thenInstructions = nullptr;
+	elseInstructions = nullptr;
+}
+
+void FCLInstruction::setSource(int32 _source) {
+	source = _source;
+}
+
+void FCLInstruction::setAdditional(int32 _additional) {
+	additional = _additional;
+}
+
+void FCLInstruction::setDestination(int32 _destination) {
```

Comment:
So, with the requirement of having class variables starting with an underscore, the typical format for methods like this is:

```C++
void FCLInstruction::setDestination(int32 destination_) {
	_destination = destination_;
}
```

Reply:
Fixed